cadccutout/cadccutout/file_helpers/fits/fits_file_helper.py

```diff
@@ -162,43 +162,53 @@ def _pixel_cutout(self, hdu, cutout_dimension):
             data=hdu.data, cutout_dimension=cutout_dimension, wcs=wcs)
 
         self._post_sanitize_header(header, cutout_result)
-        self._write_cutout(header=header, data=cutout_result.data)
-
-    def _write_primary_hdu(self, hdu_list):
-        primary_hdu = hdu_list[0]
-        if isinstance(primary_hdu, PrimaryHDU):
-            logger.debug('Writing out Primary HDU.')
-            fits.append(
-                filename=self.output_writer, header=primary_hdu.header,
-                data=None, overwrite=False, output_verify='silentfix',
-                checksum='remove')
-            logger.debug('Wrote out Primary HDU.')
+        result = fits.ImageHDU(header=header, data=cutout_result.data)
+        self._fix_header(result.header, header)
+        return result
+
+    def _fix_header(self, target, original):
+        # overcome astropy's tendency to remove keywords from original headers
+        if 'BSCALE' in original:
+            target['BSCALE'] = original['BSCALE']
+        if 'BZERO' in original:
+            target['BZERO'] = original['BZERO']
+
+    def _add_primary_hdu(self, hdu_list, result_hdu_list):
+        if isinstance(hdu_list[0], PrimaryHDU):
+            logger.debug('Appending Primary HDU.')
+            return fits.HDUList([hdu_list[0]])
         else:
-            logger.debug('HDU List does NOT contain a Primary HDU.  Skipping.')
+            logger.debug('HDU List does NOT contain a Primary HDU.'
+                         'Skipping.')
+            return result_hdu_list
 
     def _check_hdu_list(self, cutout_dimensions, hdu_list):
         has_match = False
         len_cutout_dimensions = len(cutout_dimensions)
         if len_cutout_dimensions > 0:
 
+            result_hdu_list = None
             # Check for a pixel cutout
             if isinstance(cutout_dimensions[0], PixelCutoutHDU):
                 if len_cutout_dimensions > 1:
-                    self._write_primary_hdu(hdu_list)
-
+                    result_hdu_list = \
+                        self._add_primary_hdu(hdu_list, result_hdu_list)
                 for cutout_dimension in cutout_dimensions:
                     ext = cutout_dimension.get_extension()
                     ext_idx = hdu_list.index_of(ext)
                     hdu = hdu_list[ext_idx]
 
                     # Entire extension was requested.
                     if not cutout_dimension.get_ranges():
-                        logger.debug('Writing out entire extension')
-                        self._write_cutout(header=hdu.header, data=hdu.data)
+                        logger.debug(
```

Comment:
Given my expectation of how '_add_primary_hdu' is going to be have this logic in _add_hdu is not needed.  Or, given that we have the logic in _add_hdu I think the _add_primary_hdu is not needed, except in cases where hdu_list[0] has NAXIS=0